### PR TITLE
Add plot for hyperparameter tuning

### PR DESCRIPTION
**Pull request by zenalapp**
- Function to get hyperparameters from `results$trained_model`.
- Function to combine hyperparameters for different trained models.
- Function to plot performance for a single hyperparameter (including for more than one data split if desired).

### CODE REVIEW

File: R/plot.R [hunk 1]
```diff
@@ -82,3 +82,67 @@ tidy_perf_data <- function(performance_df) {
...
+#' @param trained_model_lst list of trained models
+#'
+#' @return list: dat - dataframe of performance metric for each group of hyperparameters, params - hyperparameters tuned, metric - performance metric used 
+#' @export
+#'
+#' @examples
+#' combine_hp_performance(list(otu_mini_results1$trained_model,otu_mini_results1$trained_model))
+combine_hp_performance <- function(trained_model_lst){
+  abort_packages_not_installed("purrr")
+  dat_params <- lapply(trained_model_lst, function(x) get_hp_performance(x)) %>% purrr::transpose()
```
Comment (kelly-sovacool) [inline R/plot.R:115]: Can you think of a way to do this without purrr? That way we don't have to add it to Suggests.

**Comment by kelly-sovacool** [comment]
I'm thinking we should rename `plot_performance()` to be a bit more descriptive, since it's not immediately obvious from the name how it's different from `plot_hp_performance()`. Any ideas?

File: R/plot.R (hunk 1 above)
Comment (kelly-sovacool) [reply R/plot.R]: Not a big deal if not

File: R/plot.R [hunk 2]
```diff
@@ -181,11 +181,13 @@ combine_hp_performance <- function(trained_model_lst) {
...
   return(dat %>%
-    ggplot2::ggplot(ggplot2::aes(
-      group = {{ param_col }},
-      y = {{ metric_col }}
-    )) +
-    ggplot2::geom_boxplot())
+           dplyr::group_by({{ param_col }}) %>% 
+           dplyr::summarise(mean_AUC = mean( {{metric_col}} ), sd_AUC = sd( {{metric_col}} )) %>% 
+           ggplot2::ggplot(ggplot2::aes(x = {{ param_col }},
+                                        y = mean_AUC)) +
```
Comment (kelly-sovacool) [inline R/plot.R:189]: We can't assume AUC was the metric used. These can't be hard-coded.

Comment (kelly-sovacool) [reply R/plot.R:189]: See example here: https://github.com/SchlossLab/tidy-eval/tree/key#4-bonus-write-a-function-that-names-a-new-column-based-on-the-variable

**Comment by BTopcuoglu** [comment]
@zenalapp I changed it to dot+line plot (with sd as error bars) like the ones I had in the manuscript. I think it captures the global maxima, minima concept pretty well for tuning but we can discuss if there are better ways. It looks similar to the this figure but less prettied: https://github.com/SchlossLab/Topcuoglu_ML_mBio_2020/blob/master/submission/Figure_S2.tiff

File: R/plot.R (hunk 1 above)
Comment (zenalapp) [reply R/plot.R]: There are some other base r options:
https://stackoverflow.com/questions/15263146/revert-list-structure
e.g.
```
revert_list_str_4 <- function(ls) { # @Josh O'Brien
  # get sub-elements in same order
  x <- lapply(ls, `[`, names(ls[[1]]))
  # stack and reslice
  apply(do.call(rbind, x), 2, as.list) 
}
```

We can change it if you want.

Comment (kelly-sovacool) [reply R/plot.R]: oh that's pretty ugly, let's just keep purrr then

**Review by kelly-sovacool — APPROVED** [review]

**Pull request merged**